packages/shared/src/components/cards/WelcomePostCardHeader.tsx

```diff
@@ -4,29 +4,40 @@ import { ProfilePicture } from '../ProfilePicture';
 import SourceButton from './SourceButton';
 import PostMetadata from './PostMetadata';
 
-type WelcomePostCardHeaderProps = Pick<Post, 'author' | 'source' | 'createdAt'>;
+type WelcomePostCardHeaderProps = Pick<
+  Post,
+  'author' | 'source' | 'createdAt'
+> & { enableSourceHeader?: boolean };
 
 export const WelcomePostCardHeader = ({
   author,
   createdAt,
   source,
+  enableSourceHeader = false,
 }: WelcomePostCardHeaderProps): ReactElement => {
   return (
     <div className="flex relative flex-row gap-2 m-2 mb-3">
       <div className="relative">
-        <ProfilePicture user={author} />
+        <ProfilePicture
+          user={author}
+          size={enableSourceHeader ? 'xsmall' : 'xlarge'}
+          className={enableSourceHeader && 'top-7 -right-2.5'}
+          absolute={enableSourceHeader}
+        />
         <SourceButton
           source={source}
-          className="absolute -right-2 -bottom-2"
-          size="xsmall"
+          className={!enableSourceHeader && 'absolute -right-2 -bottom-2'}
+          size={enableSourceHeader ? 'large' : 'xsmall'}
         />
       </div>
       <div className="flex flex-col flex-1 flex-grow mr-6 ml-2 typo-footnote">
-        <span className="font-bold line-clamp-2">{author.name}</span>
+        <span className="font-bold line-clamp-2">
+          {enableSourceHeader ? source.name : author.name}
+        </span>
```

Comment:
Also `source` might be undefined, so guard with `!!source && ( ... )`

Reply:
Is this a race-condition kind of thing? I mean, in which case where we get the `source` to be `undefined`?

Reply:
Do we not have `sourceId=unknown`? In that case `source` would be `undefined` right?

Reply:
Yes, that's right. Though we never show post with `unknown` source in a modal or page, and I think for WelcomePost/FreeformPost the source is always the Squad (for the purpose of the component in the subject).

Reply:
Yeah sure, let's leave it like this here. Tnx for double checking.

Reply:
Agreed. Also would love to touch more about the topic you mentioned relating to `strictNullChecks`. Would be amazing if we can have that enabled.